src/queries.ts

```diff
@@ -0,0 +1,68 @@
+import { Pool } from 'pg'
+import { Response, Request } from 'express'
+import { config } from 'dotenv';
+
+var c = config({path: '../.env'});
+
+const pool = new Pool({
+    database: process.env.PGDATABASE,
+    user: process.env.PGUSER,
+    host: 'localhost',
+    password: process.env.PGPASSWORD, 
+    port: 5432
+})
+
+export const getTodos = (request: Request, response: Response) => {
+    pool.query('SELECT * FROM todos ORDER BY id ASC', (error, results) => {
+        if(error) {
+            throw error
+        }
+        response.status(200).json(results.rows)
+    }) 
+}
+
+export const getTodoById = (request: Request, response: Response) => {
+    const id = parseInt(request.params.id)
+    pool.query('SELECT * FROM todos WHERE id = $1', [id], (error, results) => {
+        if(error){
+            throw error;
+        }
+        response.status(200).json(results.rows)
+    })
+}
+
+export const createTodo = (request: Request, response: Response) => {
+    const {title} = request.body
+
+    pool.query('INSERT INTO todos (title) VALUES ($1)', [title], (error, results) => {
```

Comment:
What does the `$1` represent? (I'm ignorant)

Reply:
It's a numbered placeholder. So my understanding is that it is replaced with the next item in the query inside the square brackets (in this example: [title] which is pulled off of the request.body. 

If you had two values you were inserting, it would look like this ex:

`export const createTodo = (request: Request, response: Response) => {
const {title, description} = request.body

pool.query('INSERT INTO users (title, description) VALUES ($1, $2)', [title, description], (error, results) => {
    if (error) {
      throw error
    }
response.status(201).send(`Todo added with ID: ${results}`)
  })
}`

So the $1 would correspond to title, and the $2 would correspond to the description.

Reply:
@KileyAdams92 ah I see—I figured it was something like that but didn't see the correlation until you pointed that out 👌